neo/Network/P2P/Payloads/GetFullBlocksPayload.cs

```diff
@@ -0,0 +1,38 @@
+using Neo.IO;
+using System;
+using System.IO;
+
+namespace Neo.Network.P2P.Payloads
+{
+    public class GetFullBlocksPayload : ISerializable
+    {
+        public const ushort MaxBlocksCount = 500;
+        public uint IndexStart;
```

Comment:
the whole concept of this PR is to avoid having to know any hashes in order to request blocks.

Reply:
You always know the current hash, because you start from the genesis block

Reply:
Using the current hash you only increase the packet 16 bytes more, and is more reasonable for me.
`NextFullBlocks` `{CurrentHash, Amount}`

Reply:
If we change this to hash then it becomes exactly the same as the current [GetBlocksPayload](https://github.com/neo-project/neo/blob/2ca5b6b62d8870be5cec4adbfe662aaefd8cb6b5/neo/Network/P2P/Payloads/GetBlocksPayload.cs) improving nothing. 

We could instead change https://github.com/neo-project/neo/blob/2ca5b6b62d8870be5cec4adbfe662aaefd8cb6b5/neo/Network/P2P/ProtocolHandler.cs#L158
to not return hashes but return the full blocks e.g.
```C#
        private void OnGetBlocksMessageReceived(GetBlocksPayload payload)
        {
            UInt256 hash = payload.HashStart;
            int count = payload.Count < 0 ? InvPayload.MaxHashesCount : Math.Min(payload.Count, InvPayload.MaxHashesCount);
            TrimmedBlock state = Blockchain.Singleton.Store.GetBlocks().TryGet(hash);
            if (state == null) return;
            for (uint i = state.Index; i <= state.Index+count; i++)
            {
                Block block = Blockchain.Singleton.Store.GetBlock(i);
                if (block == null)
                    break;

                if (bloom_filter == null)
                {
                    Context.Parent.Tell(Message.Create(MessageCommand.Block, block));
                }
                else
                {
                    BitArray flags = new BitArray(block.Transactions.Select(p => bloom_filter.Test(p)).ToArray());
                    Context.Parent.Tell(Message.Create(MessageCommand.MerkleBlock, MerkleBlockPayload.Create(block, flags)));
                }
            }
        }
```

I think that's at least 1 step forward. But there's room for 2 by not requiring any hash knowledge to request a block.

Reply:
@shargon, I think that since we have block finality we do not need the hash anymore.
Furthermore, we need to keep emphasizing this, that we have a stable consensus (perhaps, the first in the history of the Blockchain to really achieve Block Finality) in a fully distributed scenario with CN spread around the world.

I think that we need to keep the option to ask for a `hash`, however, this do not need to be default for NEO.
In a future scenario, if more than 66% are byzantine and we get forks we could establish other seed nodes rules for defining which block to follow.

Reply:
> , I think that since we have block finality we do not need the hash anymore.
Furthermore, we need to keep emphasizing this, that we have a stable consensus (perhaps, the first in the history of the Blockchain to really achieve Block Finality in a fully distributed scenario with CN spread around the world).

Yes, but we cannot pretend to think that no more mistakes will appear in the future, it is better to anticipate and take action. Assume that we will never have a fork problem again, it is not the right way, we must prevent these situations to be more resilient.

Reply:
Shargon, I think we can leave the done and the option, for nodes that want that, but not as default anymore.

Indexes are enough to us.

Reply:
Always you will have at least the first hash, so my proposal is to change the behaviour to:

- CurrentHash && How many block do you want

Why?
- The improve is the same, only few more bytes
- More resilient

Reply:
> Always you will have at least the first hash, so my proposal is to change the behaviour to:
> 
> * CurrentHash && How many block do you want
> 
> Why?
> 
> * The improve is the same, only few more bytes
> * More resilient

@shargon can you confirm the approach described here https://github.com/neo-project/neo/pull/1138#discussion_r334882378?

Which is in summary:
- discard the whole `GetFullBlocks` payload
- change `OnGetBlocksMessageReceived()` to return full blocks

Reply:
Yes, I prefer to return full blocks on `OnGetBlocksMessageReceived` (but is just my opinion).

Reply:
I am marking this as Resolved for facilitating review, ok @shargon?